### PR TITLE
fs/promises: mkdtemp, createReadStream, writeFile, appendFile, readlink, symlink, unlink, readdir, and realpath

### PR DESCRIPTION
**Pull request by niik**
## Description

Based on #14078, this replaces our use of the following `fs-extra` functions: `mkdtemp`, `createReadStream`, `writeFile`, `appendFile`, `readlink`, `symlink`, `unlink`, `readdir`, and `realpath`.

`createReadStream` is the exact same function as that from `fs` and the others are [promisified versions](https://github.com/jprichardson/node-fs-extra/blob/7b12b058e27df560ba777756f38f977662c23750/lib/fs/index.js#L8-L40) of their fs counterparts and can be safely replaced.

## Release notes

Notes: no-notes

### CODE REVIEW

File: app/src/ui/add-repository/gitignores.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import * as Path from 'path'
-import { readdir, writeFile } from 'fs-extra'
-import { readFile } from 'fs/promises'
+import { readdir } from 'fs/promises'
+import { readFile, writeFile } from 'fs/promises'
```
Comment (sergiou87) [inline app/src/ui/add-repository/gitignores.ts]: All in one import maybe?
```suggestion
import { readdir, readFile, writeFile } from 'fs/promises'
```

**Review by sergiou87 — COMMENTED** [review]
Looks great!! Only found one minor thing 😄

File: app/src/ui/add-repository/gitignores.ts (hunk 1 above)
Comment (niik) [reply app/src/ui/add-repository/gitignores.ts]: Love it!

**Review by sergiou87 — APPROVED** [review]
:shipit: ❗

**Pull request merged**